File: 프로그래머스/lv2/42885. 구명보트/구명보트.java
import java.util.LinkedList;
import java.util.Deque;
import java.util.Arrays;
class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);

        Deque<Integer> dq = new LinkedList<>();
        int count =0;

        for(int i : people){
            dq.add(i);
        }

        while(true){
            if(dq.isEmpty()){
                break;
            }
            if(dq.size() == 1){
                count++;
                break;
            }
            if(dq.getFirst() + dq.getLast() <= limit){
                dq.removeFirst();
                dq.removeLast();
                count++;
            }
            else {
                dq.removeLast();
                count++;
            }
        }
        return count;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let answer = 0;
    people.sort((a, b) => b - a);
    
    let left = 0;
    let right = people.length - 1;
    
    while (left <= right) {
        if (people[left] + people[right] <= limit) {
            right--;
        }
        
        left++;
        answer++;
    }
    
    return answer;
}